File: FlightTrackerAPI/Database/Models/Flight.cs
using FlightTrackerAPI.Database.Models;
using System.ComponentModel.DataAnnotations;

namespace Database.Models {
    internal class Flight {

        [Key]
        public required string Id {
            get; set;
        }

        public string? FlightNumber {
            get; set;
        }

        public string? Callsign {
            get; set;
        }

        public int? AircraftId {
            get; set;
        }

        public Aircraft? Aircraft {
            get; set;
        }

        public int? AirlineId {
            get; set;
        }

        public Airline? Airline {
            get; set;
        }

        public int? OriginId {
            get; set;
        }

        public Airport? Origin {
            get; set;
        }

        public int? DestinationId {
            get; set;
        }

        public Airport? Destination {
            get; set;
        }

        public double? Altitude {
            get; set;
        }

        public double? GroundSpeed {
            get; set;
        }

        public double? Heading {
            get; set;
        }

        public long? STD {
            get; set;
        }

        public long? ETD {
            get; set;
        }

        public long? ATD {
            get; set;
        }

        public long? STA {
            get; set;
        }

        public long? ETA {
            get; set;
        }

        public long? ATA {
            get; set;
        }

        public DateTime Timestamp {
            get; set;
        }

        public ActiveFlight? ActiveFlight {
            get; set;
        }

    }
}

File: types/flight.ts
import { Aircraft } from "./aircraft.ts";
import { Airline } from "./airline.ts";
import { Airport } from "./airport.ts";

export type Flight = {
  id: string;
  callsign: string;
  flightNumber: string;
  altitude: number;
  groundSpeed: number;
  heading: number;
  departureTime: {
    scheduled: number;
    estimated: number;
    actual: number;
  };
  arrivalTime: {
    scheduled: number;
    estimated: number;
    actual: number;
  };
  timestamp: number;
  aircraft: Aircraft;
  airline: Airline;
  origin: Airport;
  destination: Airport;
};
